File: WinForms/VB/HtmlMainDemo/Modules/ListBox/ListBoxModule.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports DevExpress.DXperience.Demos
Imports DevExpress.HTML.Demos.Helpers

Namespace DevExpress.HTML.Demos

    Public Partial Class ListBoxModule
        Inherits TutorialControlBase

        Private employeesOnline As Dictionary(Of Integer, Boolean)

        Public Sub New()
            InitializeComponent()
            listBoxTemplateCodeViewer1.Fill(listBoxControl, htmlContentPopup)
            listBoxControl.ShowFocusRect = False
            listBoxControl.DataSource = DataHelper.Employees
            AddHandler listBoxControl.CustomizeItem, AddressOf ListBoxControl1_CustomizeItem
            AddHandler listBoxControl.HtmlElementMouseClick, AddressOf ListBoxControl1_HtmlElementMouseClick
            employeesOnline = Enumerable.Range(0, listBoxControl.ItemCount).ToDictionary(Function(x) x, Function(x) TutorialConstants.Random.NextDouble() >= 0.5)
        End Sub

        Private Sub ListBoxControl1_HtmlElementMouseClick(ByVal sender As Object, ByVal e As XtraEditors.ListBoxHtmlElementMouseEventArgs)
            If Equals(e.ElementId, "info") Then
                Dim size = ScaleDPI.ScaleSize(New Size(352, 360))
                Dim location = New Point(e.Bounds.Right - ScaleDPI.ScaleHorizontal(6), e.Bounds.Y + ScaleDPI.ScaleHorizontal(8) - (size.Height - e.Bounds.Height) \ 2)
                Dim tooltipScreenBounds = listBoxControl.RectangleToScreen(New Rectangle(location, size))
                htmlContentPopup.DataContext = e.Item
                htmlContentPopup.Show(Me, tooltipScreenBounds)
            End If
        End Sub

        Private Sub ListBoxControl1_CustomizeItem(ByVal sender As Object, ByVal e As XtraEditors.CustomizeTemplatedItemEventArgs)
            Dim statusBadge = e.HtmlElement?.FindElementById("status")
            Dim online As Boolean = employeesOnline(e.Index)
            If statusBadge IsNot Nothing AndAlso online Then statusBadge.Style.SetBackgroundColor("@Green")
        End Sub
    End Class
End Namespace

File: WinForms/CS/HtmlMainDemo/Modules/ListBox/ListBoxModule.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.DXperience.Demos;
using DevExpress.Utils;
using DevExpress.HTML.Demos.Helpers;

namespace DevExpress.HTML.Demos {
    public partial class ListBoxModule : TutorialControlBase {
        Dictionary<int, bool> employeesOnline;

        public ListBoxModule() {
            InitializeComponent();
            listBoxTemplateCodeViewer1.Fill(listBoxControl, htmlContentPopup);
            listBoxControl.ShowFocusRect = false;
            listBoxControl.DataSource = DataHelper.Employees;

            listBoxControl.CustomizeItem += ListBoxControl1_CustomizeItem;
            listBoxControl.HtmlElementMouseClick += ListBoxControl1_HtmlElementMouseClick;

            employeesOnline = Enumerable.Range(0, listBoxControl.ItemCount).ToDictionary(x => x, x => TutorialConstants.Random.NextDouble() >= 0.5);
        }

        void ListBoxControl1_HtmlElementMouseClick(object sender, XtraEditors.ListBoxHtmlElementMouseEventArgs e) {
            if(e.ElementId == "info") {
                var size = ScaleDPI.ScaleSize(new Size(352, 360));
                var location = new Point(
                    e.Bounds.Right - ScaleDPI.ScaleHorizontal(6),
                    e.Bounds.Y + ScaleDPI.ScaleHorizontal(8) - (size.Height - e.Bounds.Height) / 2);
                var tooltipScreenBounds = listBoxControl.RectangleToScreen(new Rectangle(location, size));

                htmlContentPopup.DataContext = e.Item;
                htmlContentPopup.Show(this, tooltipScreenBounds);
            }
        }

        void ListBoxControl1_CustomizeItem(object sender, XtraEditors.CustomizeTemplatedItemEventArgs e) {
            var statusBadge = e.HtmlElement?.FindElementById("status");
            bool online = employeesOnline[e.Index];
            if(statusBadge != null && online)
                statusBadge.Style.SetBackgroundColor("@Green");
        }
    }
}
